Update testing configuration

rydente opened the pull request:
Include Mocha.js style configuration and expectations library so we can
run tests without conflicting with our linter and so we can write
human-readable specifications.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

Commit:
Update testing configuration

Include Mocha.js style configuration and expectations library so we can
run tests without conflicting with our linter and so we can write
human-readable specifications.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

wbprice commented:
You can also use `eslint-config-trails`'s test config https://www.npmjs.com/package/eslint-config-trails#testing

Commit:
Incorporate latest changes

Merge remote-tracking branch 'upstream/develop' into
testing-configuration to integrate the testing configuration with the
latest implementation.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

Commit:
Incorporate latest changes

Merge remote-tracking branch 'upstream/develop' into
testing-configuration to integrate the testing configuration changes
with the latest implementation.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

Commit:
Define style guide for test code

Use the Trails.js test style guide for the OkCandidate tests so we can
ensure the test code stays up to date with the rest of the style guide,
but also doesn't give us warnings which may only apply to our test code.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>

rydente commented:
Hey @wbprice, this one is ready for review again.  I went with your advice for the testing setup.

Commit:
Remove Mocha ESLint plugin

Remove the Mocha ESLint plugin so we can only install what we need to
work on this project.

Signed-off-by: Ryan Y <ryayak1460@protingumas.com>
